packages/host/app/components/operator-mode/code-submode/spec-preview.gts

```diff
@@ -466,17 +470,41 @@ export default class SpecPreview extends GlimmerComponent<Signature> {
   @service private declare recentFilesService: RecentFilesService;
   @service private declare specPanelService: SpecPanelService;
   @service private declare store: StoreService;
+  @tracked private search: ReturnType<getCards<Spec>> | undefined;
+
+  // We need a way to shield ourselves from the fact that ultimately the
+  // ModuleContentsResource's declarations are not strict equal after a loader reset,
+  // but still structurally the same (like a strictEqual vs deepEqual). This value
+  // ultimately is consumed by the search resource, and even though the value never
+  // changes from a deepEqual perspective, because of reactivity the search thunk is
+  // triggered because it is not equivalent in a strict equality sense. So we use a
+  // simple resource to cache the string and reactivity is only triggered when the
+  // export name changes as opposed to being triggered because of strict equality
+  // change in the selectedDeclaration arg.
+  @use private selectedDeclarationExportName = resource(() => {
+    let exportName = new TrackedObject<{ value: string | undefined }>({
+      value: undefined,
+    });
+    (async () => {
+      await Promise.resolve(); // buffer 1 microtask to prevent re-render cycles
+      if (this.args.selectedDeclaration?.exportName !== exportName.value) {
+        exportName.value = this.args.selectedDeclaration?.exportName;
+      }
+    })();
+    return exportName;
+  });
```

Comment:
This is how we adjust the reactivity to work in a deeply equal sense vs strictly equal. in this case we really only consume the export name of the selected declaration that was triggering the run away search, so i didn't opt to use `lodash.isEqual()` on the entire declaration arg.